src/wp-includes/class-wp-user.php

```diff
@@ -106,6 +109,14 @@ class WP_User {
 	 */
 	private $site_id = 0;
 
+	/**
+	 * Flag for if capability is loaded.
+	 *
+	 * @since 6.9.0
+	 * @var bool
+	 */
+	private $loaded_caps = false;
+
```

Comment:
Minor point: Instead of introducing a new `$loaded_caps` member variable, the `$caps` member could be typed as `array<string, bool>|null` instead of `array<string, bool>`. It could be default `null` whenever wanting to check if whether the caps have been loaded, you can just check if `isset( $this->caps )`. This would eliminate having to look at two different member variables.

```suggestion
```